prover/src/trace/mod.rs

```diff
@@ -89,8 +92,7 @@ pub trait Trace: Sized {
     fn validate<A, E>(
         &self,
         air: &A,
-        aux_segment: Option<&ColMatrix<E>>,
-        aux_rand_elements: &AuxTraceRandElements<E>,
+        aux_trace_with_metadata: Option<&AirAuxTraceWithMetadata<A, E>>,
```

Comment:
Should we also try to verify the auxiliary proof somewhere in this function?

Reply:
Created https://github.com/facebook/winterfell/issues/278